mipengine/node_exceptions.py

```diff
@@ -105,6 +107,24 @@ def __init__(self, node_id: str, data_model: str):
         self.message = f"Data model '{self.data_model}' is not available in node: '{self.node_id}'."
 
 
+class IncompatibleCDEs(Exception):
+    """
+    Exception raised when nodes have the same data model but with different CDEs.
+
+    Attributes:
+        cdes1 -- incompatible common data elements
+        cdes2 -- incompatible common data elements
+        message -- explanation of the error
+    """
+
+    def __init__(
+        self, cdes1: Dict[str, CommonDataElement], cdes2: Dict[str, CommonDataElement]
```

Comment:
Did you test throwing and catching this exception? Also the message should be tested that it has the proper format.

Read the description above about problems that may arise from celery exceptions, if you haven't already.